documentation/cdi/tutorial-cdi-instantiated-beans.asciidoc

```diff
@@ -1,36 +1,25 @@
 ---
-title: Components instantiated by the framework
+title: Using CDI Beans in Instantiated Components
 order: 2
 layout: page
 ---
 
-ifdef::env-github[:outfilesuffix: .asciidoc]
+= Using CDI Beans in Instantiated Components
 
-= Components instantiated by the framework
+When instantiated by Vaadin Flow, most components become managed beans. This means they are fully-fledged CDI contextual instances: all CDI features are usable when they are queried from the CDI `BeanManager` by the https://vaadin.com/directory/component/vaadin-cdi/[Vaadin CDI] add-on.
 
-Most of the components instantiated by the Vaadin framework become managed beans ( full fledged CDI contextual instances ).
-They are queried from CDI `BeanManager` by the add-on, and all of the CDI features are usable.
+The add-on looks up the CDI bean by type (component class) and `@Any`. 
 
-[NOTE]
-Add-on looks up the CDI bean by type ( component class ), and `@Any`.
-
-[NOTE]
-When type is not found as a CDI bean
-( for example ambiguous, or does not have a no-arg public constructor ),
-instantiation falls back to the default Vaadin behavior ( instantiated as a POJO ).
-After the instantiation, dependency injection is performed.
-Injects work, but other CDI features are not, because instantiated component is not a contextual instance.
+If the type is not found as a CDI bean - for example, because it is ambiguous or does not have a no-arg public constructor - instantiation falls back to the default Vaadin behavior, and the component is instantiated as a POJO. Dependency injection is performed after instantiation and injects still work, but other CDI features do not. The reason is that the instantiated component is not a contextual instance.
 
 [NOTE]
-`@PreDestroy` of `@Dependent` beans instantiated by the framework are not run.
-Since it is the default context, it is about all beans without an explicit context.
+`@PreDestroy` or `@Dependent` beans instantiated by Vaadin Flow are not run. Since it is the default context, it is about all beans without an explicit context.
```

Comment:
@kumm, what does this mean? Any ideas how to clarify this note?

Reply:
Normally `@PreDestroy` of a Dependent bean is called automatically when the normal scoped bean that it depends on (the normal scoped bean that directly or indirectly `@Inject`s it) is destroyed. It is the concept of the creational context. When the creational context is destroyed `@PreDestroy` is called.

When instantiating it manually there is no such bean, but we get the creational context, and can call destroy on our own. Since we can't determine when to call destroy ( for example calling it on detach would cause problems on re-attach ), we don't call destroy, even we don't store the reference of the creational context.

Reply:
Maybe something like this: "It is strictly a dependent object of some other object. Since the other object is unknown when instantiating it, it's `@PreDestroy` lifecycle event is undefined. Please refer to lifecycle of the `@Dependent` scope". ( like https://docs.jboss.org/weld/reference/latest/en-US/html/scopescontexts.html#_the_dependent_pseudo_scope)

Reply:
Still not very clear. How about something like `Methods annotated with @PreDestroy in Dependent beans instantiated by the framework are not run.` and leaving out the second phrase? @mstahv @kumm

Reply:
It's ok, that's what happens.

Reply:
- Changed the note in Using CDI Beans in Instantiated Components to read "Methods annotated with `@PreDestroy` in `Dependent` beans instantiated by the framework are not run."